### DISCUSSION TITLE
One server or distributed setup?

### DISCUSSION THREAD

**Discussion by ddd793** [mynaparrot/plugNmeet-server#464] [General]
I am using the plugNmeet in a virtual machine running Docker containers (redis, mariadb, livekit, livekit-ingress, plugnmeet-api).

Virtual Machine Configuration: 
Processor - 4 core of Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz
RAM - 8 Gb
SSD drive 

Would it be beneficial to distribute Docker containers across different virtual machines? For example, running a database on a separate machine could potentially improve stability?

How many simultaneous users should I split into different servers to ensure optimal performance?

**Comment by jibon57** [comment]
Separating services into different machine will always give better results. Specially separating hosting of recorder & livekit. Having dedicated server increase the ability of the application. But you can always test before taking decision based on number of users you are going to host.

**Reply by yousofkakhki** [reply]
Hi there. 
i wanted to actually host a webinar consisting of 1000 concurrent users. a teacher normally share webcam and mic to other 999 students. students should have the ability to raise hand and ask question once in a while with mic and webcam.
i have noticed that a room is only served through one livekit instance.
how to achieve this. i got some errors with nats and plugnmeet when load testing more than 400 users on pretty powerful machine. 48 core vds with 96 gb of ram and even the resources weren't  exhausted.
